packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,516 @@
+  }
+
+  public getAbsFees(customFeeRate?: FeeRateSatsPerVByte): string {
+    const feeRate = customFeeRate || this.targetFeeRate;
+    const totalSize =
+      this.parentPsbt.PSBT_GLOBAL_INPUT_COUNT + this.estimateVsize();
+    return new BigNumber(feeRate).multipliedBy(totalSize).toString();
+  }
+
+  /**
+   * Creates the initial child transaction.
+   *
+   * This method sets up the basic structure of the child transaction by:
+   * 1. Adding inputs from the parent transaction's spendable outputs
+   * 2. Calculating the initial fee based on the estimated size and desired fee rate
+   * 3. Creating an output that spends the remaining amount minus the fee
+   *
+   * The method ensures that the resulting output is above the dust threshold.
+   *
+   * @private
+   * @throws {Error} If the resulting output would be dust
+   */
+  private createChildTransaction(): void {
+    for (const outputIndex of this.spendableOutputs) {
+      const output = {
+        script: Buffer.from(
+          this.parentPsbt.PSBT_OUT_SCRIPT[outputIndex],
+          "hex",
+        ),
+        amount: new BigNumber(
+          this.parentPsbt.PSBT_OUT_AMOUNT[outputIndex].toString(),
+        ),
+      };
+
+      const input = {
+        previousTxId: this.parentPsbt.PSBT_IN_PREVIOUS_TXID[0], // Assuming parent tx has at least one input
+        outputIndex: outputIndex,
+        witnessUtxo: {
+          script: output.script,
+          amount: output.amount.toNumber(), // Changed from 'value' to 'amount'
+        },
+      };
+
+      this.childPsbt.addInput(input);
+    }
+    const totalInputValue = new BigNumber(this.totalInputValue);
+    const estimatedFee = new BigNumber(this.estimatedRequiredFee);
+
+    const outputAmount = totalInputValue.minus(estimatedFee);
+
+    if (outputAmount.isLessThan(this.dustThreshold)) {
+      throw new Error("CPFP transaction would create a dust output");
+    }
+
+    const outputScript = createOutputScript(
+      this.destinationAddress,
+      this.network,
+    );
+
+    const output = {
+      script: outputScript,
+      amount: outputAmount.toNumber(),
+    };
+    this.childPsbt.addOutput(output);
+  }
+
+  private addAdditionalInputsIfNeeded(): void {
+    const totalInput = new BigNumber(this.totalInputValue);
+    const required = new BigNumber(this.requiredAmount);
+    if (totalInput.isLessThan(required)) {
+      this.addAdditionalInputs(required.minus(totalInput));
+    }
+  }
+  /**
+   * Adds additional inputs to the child transaction if needed.
+   *
+   * This method is called when the inputs from the parent transaction are not
+   * sufficient to cover the desired fee increase. It adds UTXOs from the
+   * additionalUtxos option until the required amount is met or the maximum
+   * number of additional inputs is reached.
+   *
+   * @private
+   * @throws {Error} If there are insufficient funds in additional UTXOs
+   */
+  private addAdditionalInputs(requiredAmount: BigNumber): void {
```

Comment:
should the inputs be sorted? That way you use the largest first hopefully avoiding adding an unnecessarily large number of inputs.

Reply:
This class is depreciated now